src/audio/audio_manager.cpp

```diff
@@ -16,17 +21,74 @@ AudioManager::AudioManager() : muted(true) {
         if (!music.openFromFile(filename)) {
             std::cerr << "AudioManager: Cannot open " << filename << std::endl;;
         }
-        music.setLoop(true);
-        if (!muted)
+
+        music.setVolume(d.volume);
+        music.setLoop(d.loop);
+
+        if (!muted) {
             music.play();
+        }
+    });
+
+    events::sound_effects_event.connect([&](const events::AudioData &d) {
+        std::string filename = d.filename;
+
+        sounds[filename]->setVolume(d.volume);
+        sounds[filename]->setLoop(d.loop);
+
+        if (!muted) {
```

Comment:
How about

```c++
if (d.loop) {
  sounds[filename]->play();
  sounds[filename]->pause();
}

if (muted) return;

sounds[filename]->play();
```

It's a lot easier to understand IMO because i'm not saying "okay, if it's muted and not looping..." in my head.